drivers/sensor/fdc2x1x/fdc2x1x.c

```diff
@@ -0,0 +1,1010 @@
+	struct fdc2x1x_data *data = dev->data;
+
+	data->pm_state = DEVICE_PM_LOW_POWER_STATE;
+#endif
+
+	return ret;
+}
+
+#ifdef CONFIG_PM_DEVICE
+/**
+ * Reinitialize device after exiting shutdown mode
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int fdc2x1x_restart(const struct device *dev)
+{
+	int ret;
+
+	k_sleep(K_MSEC(100));
+
+	ret = fdc2x1x_init_config(dev);
+	if (ret) {
+		LOG_ERR("Reinitializing failed");
+		return ret;
+	}
+
+#ifdef CONFIG_FDC2X1X_TRIGGER
+	struct fdc2x1x_data *data = dev->data;
+
+	ret = fdc2x1x_reg_write_mask(dev, FDC2X1X_ERROR_CONFIG,
+				     data->int_config, data->int_config);
+	if (ret) {
+		LOG_ERR("Reinitializing trigger failed");
+		return ret;
+	}
+#endif
+
+	return 0;
+}
+
+/**
+ * Enable/disable Shutdown Mode
+ * @param dev - The device structure.
+ * @param enable - True = enable shutdown, false = disable shutdown
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int fdc2x1x_set_shutdown(const struct device *dev, bool enable)
+{
+	struct fdc2x1x_data *data = dev->data;
+	const struct fdc2x1x_config *cfg = dev->config;
+	int ret = 0;
+
+	gpio_pin_set(data->gpio, cfg->sd_pin, enable);
+
+	if (!enable) {
+		ret = fdc2x1x_restart(dev);
+	}
+
+	return ret;
+}
+
+/**
+ * Set the Device Power Management State.
+ * @param dev - The device structure.
+ * @param pm_state - power management state
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int fdc2x1x_set_pm_state(const struct device *dev,
+				uint32_t pm_state)
+{
+	int ret;
+	struct fdc2x1x_data *data = dev->data;
+	const struct fdc2x1x_config *cfg = dev->config;
+
+	switch (pm_state) {
+	case DEVICE_PM_ACTIVE_STATE:
+		if (data->pm_state == DEVICE_PM_OFF_STATE) {
+			ret = fdc2x1x_set_shutdown(dev, false);
+			if (ret) {
+				return ret;
+			}
+		}
+
+		ret = fdc2x1x_set_op_mode(dev, FDC2X1X_ACTIVE_MODE);
+		if (ret) {
+			return ret;
+		}
+		data->pm_state = DEVICE_PM_ACTIVE_STATE;
+
+		break;
+	case DEVICE_PM_LOW_POWER_STATE:
+		if (data->pm_state == DEVICE_PM_OFF_STATE) {
+			ret = fdc2x1x_set_shutdown(dev, false);
+			if (ret) {
+				return ret;
+			}
+		}
+		ret = fdc2x1x_set_op_mode(dev, FDC2X1X_SLEEP_MODE);
+		if (ret) {
+			return ret;
+		}
+		data->pm_state = DEVICE_PM_LOW_POWER_STATE;
+
+		break;
+	case DEVICE_PM_OFF_STATE:
+		if (cfg->sd_name) {
+			ret = fdc2x1x_set_shutdown(dev, true);
+			data->pm_state = DEVICE_PM_OFF_STATE;
+		} else {
+			LOG_ERR("SD pin not defined");
+			ret = -EINVAL;
+		}
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return ret;
+}
+
+static int fdc2x1x_device_pm_ctrl(const struct device *dev,
+				  uint32_t ctrl_command,
+				  void *context, device_pm_cb cb, void *arg)
+{
+	struct fdc2x1x_data *data = dev->data;
+	uint32_t new_state;
+	int ret = 0;
+
+	if (ctrl_command == DEVICE_PM_SET_POWER_STATE) {
+		new_state = *(uint32_t *)context;
+
+		if (new_state != data->pm_state) {
+
```

Comment:
Nit: unnecessary whitespace.

Reply:
> Nit: unnecessary whitespace.

Still here.

Reply:
sorry, my bad. Removed the whitespace now